elide-datastore/elide-datastore-multiplex/src/main/java/com/yahoo/elide/datastores/multiplex/MultiplexTransaction.java

```diff
@@ -200,43 +202,43 @@ public void updateToManyRelation(DataStoreTransaction relationTx,
     public void updateToOneRelation(DataStoreTransaction relationTx, Object entity,
                                     String relationName, Object relationshipValue, RequestScope scope) {
         relationTx = getRelationTransaction(entity, relationName);
-        DataStoreTransaction entityTransaction = getTransaction(entity.getClass());
+        DataStoreTransaction entityTransaction = getTransaction(getType(entity.getClass()));
         entityTransaction.updateToOneRelation(relationTx, entity, relationName, relationshipValue, scope);
     }
 
     @Override
     public Object getAttribute(Object entity, Attribute attribute, RequestScope scope) {
-        DataStoreTransaction transaction = getTransaction(entity.getClass());
+        DataStoreTransaction transaction = getTransaction(getType(entity.getClass()));
         return transaction.getAttribute(entity, attribute, scope);
     }
 
     @Override
     public void setAttribute(Object entity, Attribute attribute, RequestScope scope) {
-        DataStoreTransaction transaction = getTransaction(entity.getClass());
+        DataStoreTransaction transaction = getTransaction(getType(entity.getClass()));
```

Comment:
Everywhere we call entity.getClass - we need to use the EntityDictionary.getTYpe instead.